sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -808,9 +808,24 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
     assertNotBucketed("jdbc")
     // connectionProperties should override settings in extraOptions.
     this.extraOptions ++= connectionProperties.asScala
-    // explicit url and dbtable should override all
-    this.extraOptions ++= Seq("url" -> url, "dbtable" -> table)
-    format("jdbc").save()
+    // explicit url should override all
+    this.extraOptions += "url" -> url
+
+    import df.sparkSession.sessionState.analyzer.NonSessionCatalogAndIdentifier
+    import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
+
+    val session = df.sparkSession
+    this.source = "jdbc"
+
+    session.sessionState.sqlParser.parseMultipartIdentifier(table) match {
+      case nameParts @ NonSessionCatalogAndIdentifier(catalog, tableIdentifier) =>
+        saveAsTable(catalog.asTableCatalog, tableIdentifier, nameParts)
```

Comment:
Returned catalog is always a `TableCatalog`?

Reply:
Or do we need to check if `catalog` is `JDBCTableCatalog`?

Reply:
Seems to me that the catalog in `NonSessionCatalogAndIdentifier` is always a `TableCatalog`.